tests/server/test_server.py

```diff
@@ -0,0 +1,536 @@
+                        }
+                    }
+                }),
+                encrypt=self._use_encryption
+            )
+
+        def step2(payload):
+            self.assertEqual(
+                payload,
+                (3000, "No user information was found in this request.")
+            )
+
+        return self._chain_calls(self._activate_encryption_if_required, step1, step2)
+
+    @register
+    def _activate_encryption_if_required(self, _):
+        """
+        Activate encryption if this test suite needs it.
+        """
+        if self._use_encryption:
+            return self._activate_encryption(_)
+        else:
+            return None
+
+    @register
+    def _activate_encryption(self, _):
+        """
+        Activates encryption
+        """
+        return self._send_message("get_ws_server_id", None)
+
+    @register
+    def test_incorrect_user(self):
+        """
+        Ensure incorrect user is caught
+        """
+        def step1(_):
+            return self._send_message("repeat_value", {"value": "hello"}, encrypt=self._use_encryption, user_id=666)
+
+        def step2(payload):
+            self.assertEqual(
+                payload,
+                (
+                    3001,
+                    "You are not authorized to make browser integration requests. "
+                    "Please re-authenticate in your desktop application."
+                )
+            )
+
+        return self._chain_calls(self._activate_encryption_if_required, step1, step2)
+
+    @register
+    def test_unknown_command(self):
+        """
+        Ensures that a method that is not in the supported list raises an error.
+        """
+        def step1(_):
+            return self._send_message("unknown_command", None, encrypt=self._use_encryption)
+
+        def step2(payload):
+            if self._use_encryption:
+                payload = self._fernet.decrypt(payload)
+            payload = json.loads(payload)
+            self._is_error(payload, "Command unknown_command is not supported.")
+
+        return self._chain_calls(self._activate_encryption_if_required, step1, step2)
+
+    @register
+    def test_missing_method(self):
+        """
+        Ensures a public method that has a missing matching attribute raises an error.
+        """
+        def step1(_):
+            return self._send_message("missing_attribute", None, encrypt=self._use_encryption)
+
+        def step2(payload):
+            if self._use_encryption:
+                payload = self._fernet.decrypt(payload)
+            payload = json.loads(payload)
+            self._is_error(payload, "Could not find API method missing_attribute: ")
+
+        return self._chain_calls(self._activate_encryption_if_required, step1, step2)
+
+    return type(class_name, class_parents, class_attr)
+
+
+class TestEncryptedServer(unittest.TestCase):
+    """
+    Tests for various caching-related methods for api_v2.
+    """
+    __metaclass__ = TestServerBase
```

Comment:
Seems like twisted.trial has issue with unit test base classes, so we're using a metaclass instead to add basic functionality to the test classes.